datajoint/schema.py

```diff
@@ -1,15 +1,59 @@
+from operator import itemgetter
 import pymysql
 import logging
 
-from . import conn, DataJointError, NoDefinitionError
+from . import conn, DataJointError
+from datajoint.utils import to_camel_case
 from .heading import Heading
 from .base_relation import BaseRelation
-from .user_relations import Part
+from .user_relations import Part, Computed, Imported, Manual, Lookup
 import inspect
+
 logger = logging.getLogger(__name__)
 from warnings import warn
 
 
+class SchemaFactory:
+    """
+    Creates the appropriate python schema class objects from tables in a database.
+
+    SchemaFactory stores the class objects in a class dictionary and returns those
+    when prompted for the same table from the same database again. This way, the id
+    of both returned class objects is the same and comparison with python's "is" works
+    correctly.
+    """
+
+    table2class = {}
+
+    def __call__(self, database, table_name):
+        class_name = to_camel_case(table_name)
+
+        def _make_tuples(other, key):
+            raise NotImplementedError("This is an automatically created class. _make_tuples is not implemented.")
+
+        if (database, table_name) in SchemaFactory.table2class:
+            class_name, class_obj = SchemaFactory.table2class[database, table_name]
+        else:
+            if table_name.rfind('__') > 1:
```

Comment:
regular expressions allow more precise matching.  The naming convention involves more than just the prefixes.  Tables that don't follow the conventions should be ignored.